core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt

```diff
@@ -0,0 +1,172 @@
+        val defaultStorage = defaultStorage()
+        putPrefStorage(defaultStorage)
+        return defaultStorage().also(::putPrefStorage)
+      } else if (!File(storage).exists()) {
+        return defaultStorage()
+      }
+      return storage
+    }
+
+  private fun defaultStorage(): String =
+    getExternalFilesDirs(instance, null)[0]?.path ?: instance.filesDir.path
+
+  fun getPrefStorageTitle(defaultTitle: String): String? =
+    sharedPreferences.getString(PREF_STORAGE_TITLE, defaultTitle)
+
+  fun putPrefLanguage(language: String?) =
+    sharedPreferences.edit().putString(PREF_LANG, language).apply()
+
+  fun putPrefIsFirstRun(isFirstRun: Boolean) =
+    sharedPreferences.edit().putBoolean(PREF_IS_FIRST_RUN, isFirstRun).apply()
+
+  fun putPrefWifiOnly(wifiOnly: Boolean) =
+    sharedPreferences.edit().putBoolean(PREF_WIFI_ONLY, wifiOnly).apply()
+
+  fun putPrefStorageTitle(storageTitle: String?) =
+    sharedPreferences.edit().putString(PREF_STORAGE_TITLE, storageTitle).apply()
+
+  fun putPrefStorage(storage: String) {
+    sharedPreferences.edit().putString(PREF_STORAGE, storage).apply(); prefStorages.onNext(storage)
+  }
+
+  fun putPrefFullScreen(fullScreen: Boolean) =
+    sharedPreferences.edit().putBoolean(PREF_FULLSCREEN, fullScreen).apply()
+
+  fun putPrefExternalLinkPopup(externalLinkPopup: Boolean) =
+    sharedPreferences.edit().putBoolean(PREF_EXTERNAL_LINK_POPUP, externalLinkPopup).apply()
+
+  fun showIntro(): Boolean = sharedPreferences.getBoolean(PREF_SHOW_INTRO, true)
+
+  fun setIntroShown() = sharedPreferences.edit().putBoolean(PREF_SHOW_INTRO, false).apply()
+
+  var showHistoryAllBooks: Boolean
+    get() = sharedPreferences.getBoolean(PREF_SHOW_HISTORY_ALL_BOOKS, true)
+    set(prefShowHistoryAllBooks) {
+      sharedPreferences.edit().putBoolean(PREF_SHOW_HISTORY_ALL_BOOKS, prefShowHistoryAllBooks)
+        .apply()
+    }
+
+  var showBookmarksAllBooks: Boolean
+    get() = sharedPreferences.getBoolean(PREF_SHOW_BOOKMARKS_ALL_BOOKS, true)
+    set(prefShowBookmarksFromCurrentBook) = sharedPreferences.edit()
+      .putBoolean(PREF_SHOW_BOOKMARKS_ALL_BOOKS, prefShowBookmarksFromCurrentBook).apply()
+
+  val nightMode: NightModeConfig.Mode
+    get() = from(
+      sharedPreferences.getString("", null)?.toInt() ?: AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
+    )
+
+  fun nightModes(): Flowable<NightModeConfig.Mode> = nightModes.startWith(nightMode)
+
+  fun updateNightMode() = nightModes.offer(nightMode)
+
+  var hostedBooks: Set<String>?
+    get() = sharedPreferences.getStringSet(PREF_HOSTED_BOOKS, HashSet())
+    set(hostedBooks) {
+      sharedPreferences.edit().putStringSet(PREF_HOSTED_BOOKS, hostedBooks).apply()
+    }
+
+  var textZoom: Int
+    get() = sharedPreferences.getInt(TEXT_ZOOM, DEFAULT_ZOOM)
+    set(textZoom) {
+      sharedPreferences.edit().putInt(TEXT_ZOOM, textZoom).apply(); textZooms.offer(textZoom)
+    }
+
+  fun getTextZooms(): Flowable<Int> = textZooms.startWith(textZoom)
+
```

Comment:
Deleting the unused `getPrefStorages()` was a mistake, that method was necessary